src/main/java/org/ibp/api/rest/labelprinting/GermplasmLabelPrinting.java

```diff
@@ -0,0 +1,566 @@
+		germplasmType.setFields(DEFAULT_GERMPLASM_DETAILS_FIELDS);
+		labelTypes.add(germplasmType);
+
+		// Pedigree labels
+		final LabelType pedigreeType = new LabelType(pedigreePropValue, pedigreePropValue);
+		pedigreeType.setFields(DEFAULT_PEDIGREE_DETAILS_FIELDS);
+		labelTypes.add(pedigreeType);
+
+		// Names labels
+		final LabelType namesType = new LabelType(namesPropValue, namesPropValue);
+		namesType.setFields(new ArrayList<>());
+		namesType.getFields().addAll(nameTypes.stream()
+			.map(attr -> new Field(toKey(attr.getFldno()), attr.getFcode()))
+			.collect(Collectors.toList()));
+		labelTypes.add(namesType);
+
+		// Attribiutes labels
+		final LabelType attirbutesType = new LabelType(attributesPropValue, attributesPropValue);
+		attirbutesType.setFields(new ArrayList<>());
+		attirbutesType.getFields().addAll(attributeTypes.stream()
+			.map(attr -> new Field(toKey(attr.getFldno()), attr.getFcode()))
+			.collect(Collectors.toList()));
+		labelTypes.add(attirbutesType);
+
+		return labelTypes;
+	}
+
+	@Override
+	LabelsData getLabelsData(
+		final LabelsGeneratorInput labelsGeneratorInput) {
+		// Get raw data
+		final Integer searchRequestId = labelsGeneratorInput.getSearchRequestId();
+		final GermplasmSearchRequest germplasmSearchRequest = (GermplasmSearchRequest) this.searchRequestService
+			.getSearchRequest(searchRequestId, GermplasmSearchRequest.class);
+
+		final Set<String> addedColumnsPropertyIds = new HashSet<>();
+
+		labelsGeneratorInput.getFields().forEach((listOfSelectedFields) ->
+			this.addingColumnToGermplasmSearchRequest(listOfSelectedFields, addedColumnsPropertyIds)
+		);
+
+		if (!StringUtils.isBlank(labelsGeneratorInput.getSortBy()) && !addedColumnsPropertyIds.contains(labelsGeneratorInput.getSortBy())) {
+			addedColumnsPropertyIds.add(labelsGeneratorInput.getSortBy());
+		}
+
+		germplasmSearchRequest.setAddedColumnsPropertyIds(addedColumnsPropertyIds.stream().collect(Collectors.toList()));
+
+		final List<Integer> listOfGermplasmDetailsAndPedrigreeIds = new ArrayList<>();
+		listOfGermplasmDetailsAndPedrigreeIds.addAll(GERMPLASM_FIELD_IDS);
+		listOfGermplasmDetailsAndPedrigreeIds.addAll(PEDIGREE_FIELD_IDS);
+
+		final List<Integer> fieldsContainsNamesOrAttributes =
+			labelsGeneratorInput.getFields().stream().flatMap(Collection::stream).collect(Collectors.toList());
+
+		final boolean haveNamesOrAttributes =
+			fieldsContainsNamesOrAttributes.stream().anyMatch((fieldId) -> !listOfGermplasmDetailsAndPedrigreeIds.contains(fieldId));
+
+		PageRequest pageRequest = null;
+		if (!StringUtils.isBlank(labelsGeneratorInput.getSortBy())) {
+			pageRequest = new PageRequest(0, GermplasmLabelPrinting.MAX_GID_LIST_SIZE, new Sort(Sort.Direction.ASC, labelsGeneratorInput.getSortBy()));
+		}
+
+		final List<GermplasmSearchResponse> responseList =
+			this.germplasmService.searchGermplasm(germplasmSearchRequest, pageRequest, null);
+
+		Map<Integer, Map<Integer, String>> attributeValues = null;
```

Comment:
Given that checking if the maps are empty was removed in getDataRow, I suggest to initialize these two var with new HashMap